src/transcode.jl

```diff
@@ -76,9 +94,11 @@ julia> String(decompressed)
 
 ```
 """
-function Base.transcode(codec::Codec, data::ByteData)
-    input = Buffer(data)
-    output = Buffer(initial_output_size(codec, buffermem(input)))
+function Base.transcode(
```

Comment:
Sorry to bring this up this late, but would people expect a version of this to be `transcode!` by convention?

Reply:
> Sorry to bring this up this late, but would people expect a version of this to be `transcode!` by convention?

`Base.transcode` is a function owned by `Base`, and `Base.transcode!` doesn't exist. I think it would be quite surprising that this package adds `Base.transcode!`. If you instead add `TranscodingStreams.transcode!` it's also weird that those two very similar functions are defined in the same module but have different parentmodules.

It's true that often functions that mutate should have an exclamation mark, but `Base` already has plenty of counter-examples where it makes sense. For example, `Base.iterate` mutates if the input is stateful.

Reply:
Add `Transcodingstreams.transcode!`. The extended methods of `Base.transcode` should call then call that.